lib/qa/authorities/linked_data/config/search_config.rb

```diff
@@ -19,26 +19,10 @@ def supports_search?
         search_config.present?
       end
 
-      # Return search url encoding defined in the configuration for this authority
-      # if it was provided
-      # @return [Hash,NilClass] the configured search url
-      def url
-        search_config[:url]
-      end
-
       # Return search url template defined in the configuration for this authority.
-      # @return [String] the configured search url template
-      def url_template
-        url.fetch(:template)
-      end
-
-      # Return search url parameter mapping defined in the configuration for this authority.
-      # @return [Hash] the configured search url parameter mappings with variable name as key
-      def url_mappings
-        return @url_mappings unless @url_mappings.nil?
-        mappings = Config.config_value(url, :mapping)
-        return {} if mappings.nil?
-        Hash[*mappings.collect { |m| [m[:variable].to_sym, m] }.flatten]
+      # @return [Qa::IriTemplate::UrlConfig] the configured search url template
+      def url_config
+        @url_config ||= Qa::IriTemplate::UrlConfig.new(search_config[:url]) if supports_search?
```

Comment:
The Qa::IriTemplate::UrlConfig class processes everything, so #url, #url_template, and #url_mappings are obsolete.  Removed these methods and their tests.